### DISCUSSION TITLE
Golang debugging with delve

### DISCUSSION THREAD

**Discussion by matoous** [helix-editor/helix#6273] [Q&A] [answered]
Hi! Did anyone manage to make debugging for golang work? If so, would you please share how?

**Comment by oyvindsk** [comment, marked as answer]
Just started, but found this very helpful: https://overlandandseas.dev/posts/2023-7-11-debugging-go-delve-helix/

Might have to combine it with some command-line dlv action :man_shrugging: 
- https://golang.cafe/blog/golang-debugging-with-delve.html
- https://github.com/go-delve/delve/blob/master/Documentation/usage/README.md

**Reply by nexovec** [reply]
first link is broken

**Reply by mikerhodes** [reply]
New link https://overlandandseas.dev/blog/debugging-go-delve-helix/

**Comment by nexovec** [comment]
There is an edit command in delve CLI, but when I use `DELVE_EDITOR=hx`, it doesn't open the correct place in code, and I'm not sure if this is a delve issue or a helix issue.

Debugging in helix directly doesn't seem usable yet(again, specifically for go) and in vs code it's not possible to access the CLI commands directly, so I'm left without a satisfactory solution.

**Reply by nexovec** [reply]
For anyone desperately still trying to get good debugger support for go, zed is fine. Only when used with helix mode, of course :wink: .
